src/Settings.php

```diff
@@ -209,4 +209,14 @@ public function getSharedSecret() {
     return $this->sharedSecret;
   }
 
+  /**
+   * Updates webhook data.
+   *
+   * @param array $webhook
+   *   The webhook data.
+   */
+  public function updateWebhook(array $webhook): void {
```

Comment:
I feel like we need to search for another solution. This would violate the purpose of this class, which is being an immutable object.

Reply:
One its attributes of an immutable object is to preserve state and avoid any side effects.